CIP-XXXX/README.md

```diff
@@ -0,0 +1,120 @@
+
+One apparent alternative choice would be to remove the HFC wrapper entirely.
+However, this would reduce modularity.
+Because the HFC wrapper controls the prefix all of the node's codecs, the wrapped Byron and Shelley codecs can exist completely independently from each other---the HFC wrapper completely disambiguates them.
+The same is true for the later hard forks to Allegra, Mary, Alonzo, and so on: each one is assigned a unique value for the HFC tag.
+While the wrapper allows the wrapped codecs to be independent, it doesn't force them to be.
+Codecs are still free to share code, which has been useful and reasonable for all the relatively-incremental changes that hvae been made since the big Byron-to-Shelley hard fork.
+But a big change in the future (perhaps Ouroboros Leios?) might benefit from the independence the HFC wrapper ensures.
+
+## Follow-Up Change: Storage Layer
+
+Altering the database schema has consequences for tools (such as Mithril) that currently depend on the exact content of the node's private on-disk files --- this kind of disruption is [a known issue](https://github.com/cardano-foundation/CIPs/pull/974#issuecomment-2624935823) with the current Mithril design.
+Making the on-disk data more objective will make future disruptions to such tools less likely.
+
+In fact, on-disk blocks (and so blocks arriving via BlockFetch) are the slight exception foreshadowed in the "Motivation" section above.
+Even before Shelley, Byron blocks were already wrapped with a tag that differentiated Epoch Boundary Blocks (0) from regular blocks (1), so the new option of Shelley (2) was merely added to that existing tag instead of nesting the old Byron wrapper within the new wrapper.
+In particular, combining those two wrappers avoided needing to migrate the node's on-disk files to a new schema.
+But since this CIP requires a change to the schema, it's also an opportunity to eliminate that special case: if a schema change is already happening, the marginal overhead of nested wrappers for Byron no longer justifies the special case.
+
+None of these tags --- even the original Byron block tag --- have ever influenced hashes, so it is possible for their schema to change; it's simply never happened before.
+
+## Follow-Up Change: Transactions
+
+Removing the HFC wrapper from transactions is an improvement in two ways.
+- Just as today's HFC tags leak implementation details into the codecs, today's `cardano-cli` command-line interface leaks those same implementation details by forcing the user to pick the HFC tag for their transaction.
+  (This is not apparent as a burden for users today, since there's essentially only one CLI interface for Cardano.)
+- As explained in the "Follow-Up Change: Backwards-Compatibility Conversions" bullet above, the node doesn't actually need an HFC wrapper for incoming transactions: the current state of the mempool determines which protocol version to use when decoding the bytes of the incoming transaction.
```

Comment:
Also: it would make the life of people writing alternative tx submission clients easier, as expressed in https://github.com/IntersectMBO/ouroboros-consensus/issues/137

Reply:
Thanks, added!